src/transformers/generation/utils.py

```diff
@@ -1087,6 +1087,25 @@ def _validate_model_class(self):
                 exception_message += f" Please use one of the following classes instead: {generate_compatible_classes}"
             raise TypeError(exception_message)
 
+    def _validate_assistant(self, assistant_model):
+        if assistant_model is None or not self.config.is_encoder_decoder:
+            return
+
+        if not assistant_model.config.is_encoder_decoder:
+            attributes_to_check = ["encoder_attention_heads", "encoder_ffn_dim", "encoder_layers"]
+            attributes_to_check = [attr for attr in dir(self.config) if attr in attributes_to_check]
```

Comment:
```suggestion
            attributes_to_check = [attr for attr in dir(assistant_model.config) if attr in attributes_to_check]
```
It's okay if the main model has these attributes and the assistant doesn't, but not the other way around :) e.g. we can have an encoder-decoder model with a encoder-only model that has no encoder-related inputs